cli/flox/src/commands/services/mod.rs

```diff
@@ -142,6 +301,26 @@ fn processes_by_name_or_default_to_all<'a>(
     }
 }
 
+/// Check if all processes are stopped and shutdown `process-compose` if they
+/// are.
+///
+/// Returns true if process-compose was shutdown.
+pub fn shutdown_process_compose_if_all_processes_stopped(
+    env: &ServicesEnvironment,
+) -> Result<bool> {
+    // Process compose was not running, so we don't need to shut it down.
+    let Some(processes) = env.processes_if_expected_running().transpose()? else {
+        return Ok(true);
```

Comment:
nonblocking suggestion: I think we should only call this after checking if process-compose is responsive. It's surprising to `guard_services_running` for some subcommands and error indirectly via calling `shutdown_process_compose_if_all_processes_stopped` for others